fixed issue 17

hyperov opened the pull request:
implement pause/resume feature in recording

Commit:
Implement pause feature by adding 2 main component:
  1-make recording service a bound service to deliver pause and resume.
  2-integrate with open source library mp4 parser to append multiple
    files to one.

  //issues :
  1-file naming increment issue.

Commit:
1-file naming increment issue : fixed.
2-handle case : user press stop after pause

Commit:
delete todos regarding pause/resume feature

naXa777 committed:
+ Change package name (com.danielkim -> by.naxa)
+ update build tools
+ update dependencies

naXa777 committed:
Request app permissions at runtime (on Android 6.0 and higher)

naXa777 committed:
Integrate Crashlytics SDK to monitor the app's health

naXa777 committed:
Fix a crash (due to incorrect permissions check)

naXa777 committed:
Remove extra padding between cards on Records tab

naXa777 committed:
Change directory for storing records + refactoring

naXa777 committed:
Show error (in Snackbar) if permission is not granted

naXa777 committed:
Modernization: replace com.melnykov.fab.FloatingActionButton (deprecated) with android.support.design.widget.FloatingActionButton

naXa777 committed:
Add missing russian translations

naXa777 committed:
+ Verify that external storage is available and show error (if it's not)
+ Do not start chronometer if the recording is not going to be saved (because of I/O error)

naXa777 commented on app/src/main/java/com/danielkim/soundrecorder/RecordingService.java:
```diff
@@ -142,6 +204,60 @@ public void stopRecording() {
+//            finalMovie.addTrack(new AppendTrack((Track) tracks));
+//        } catch (IOException e) {
+//            e.printStackTrace();
+//        }
```
please remove "dead code"

naXa777 commented on app/src/main/java/com/danielkim/soundrecorder/fragments/RecordFragment.java:
```diff
@@ -151,29 +175,34 @@ public void onChronometerTick(Chronometer chronometer) {
+            mRecordingService.pauseRecording();
             mPauseButton.setCompoundDrawablesWithIntrinsicBounds
                     (R.drawable.ic_media_play ,0 ,0 ,0);
             mRecordingPrompt.setText((String)getString(R.string.resume_recording_button).toUpperCase());
```
bug: change `mPauseButton` instead of `mRecordingPrompt`. set `mRecordingPrompt` text to "Paused".

naXa777 commented on app/src/main/java/com/danielkim/soundrecorder/fragments/RecordFragment.java:
```diff
@@ -151,29 +175,34 @@ public void onChronometerTick(Chronometer chronometer) {
+            mRecordingService.resumeRecording();
             mPauseButton.setCompoundDrawablesWithIntrinsicBounds
                     (R.drawable.ic_media_pause ,0 ,0 ,0);
             mRecordingPrompt.setText((String)getString(R.string.pause_recording_button).toUpperCase());
```
bug: change `mPauseButton` instead of `mRecordingPrompt`. add `mRecordingPrompt.setText(getString(R.string.record_in_progress));`

naXa777 commented on app/src/main/res/values/strings.xml:
```diff
@@ -13,6 +13,8 @@
     <string name="toast_recording_start">Recording started</string>
     <string name="toast_recording_finish">Recording saved to</string>
+    <string name="toast_recording_paused">Recording paused</string>
+    <string name="toast_recording_resumed">Recording resumed</string>
```
The string `toast_recording_resumed` is not used anywhere. Did you forget to add a toast?

naXa777 committed:
Update strings

naXa777 committed:
Update Gradle build plugin to the latest version

naXa777 committed:
Warn when file operations fail

naXa777 committed:
cleanup

naXa777 committed:
Use Android Design Support CoordinatorLayout + AppBarLayout + TabLayout to implement Material Design tabs

naXa777 committed:
Report recording task status (from Service to Activity) - this commit fixes occasional application crashes when RecordingFragment is recreated while RecordingService is recording

naXa777 committed:
Cleanup: minor changes to project hierarchy

Commit:
remove dead code

Commit:
add Toast for resumed status.
change mRecordingPrompt properly.

Commit:
change mPlayButton to FAB.
texts minor fixes

hyperov commented:
hi @naXa777 
thank you for your comments.
i fixed the bugs,and changes pause button to FAB as it seems more suitable

naXa777 committed:
Double tapping on Record button was causing Runtime exception at stop()

naXa777 committed:
Declares hardware features that are used by the application

naXa777 committed:
Request audio focus when playing a sound recording

naXa777 committed:
Minor changes:
MainActivity may be exercised by the monkey;
Update gradle build plugin.

naXa777 committed:
Respond to headset events: pause playback when headset is unplugged

naXa777 committed:
* Fix AppBar layout
* Group preferences by categories

naXa777 committed:
* Fix "black hole" bug in Android 4.1 (API 16)
* Animate indeterminate Circular ProgressBar when recording is in progress

naXa777 committed:
Add landscape mode

naXa777 committed:
Apply Material Design to all components (such as PAUSE button, EditText, Dialogs)

naXa777 committed:
Refactor > Migrate to AndroidX

naXa777 committed:
Fix layout problems

naXa777 committed:
Fix FileUriExposedException, which occurred when sharing files through intents on Nougat+

naXa777 committed:
Update Open Source Licenses

naXa777 committed:
prepare for release - app version 2.0

naXa777 committed:
"Rename File" dialog - limit TextInputEditText to single line

naXa777 committed:
Update README.md

naXa777 committed:
Update .gitignore

naXa777 committed:
"Rename File" dialog - show soft keyboard automatically

naXa777 committed:
Update README.md

naXa777 committed:
Fix application crash in case if User starts recording when the microphone is in use by another app

naXa777 committed:
* Fix application crash when leaving split-screen mode in Android Nougat+
* Fix inspection warnings in order to prevent NPEs

naXa777 committed:
Update README.md - Contribution

naXa777 committed:
Remove ignored files from VCS

naXa777 committed:
Start RecordingService in foreground (show persistent notification during recording).
Fixes issues #1 and #4

naXa777 committed:
Fix high quality preference

naXa777 committed:
Fix #31 - Change time format from mm:ss to hh:mm:ss for long records

naXa777 committed:
#30 Move app icons to mipmap- folders

naXa777 committed:
Integrate with Travis CI

naXa777 committed:
Setup Fabric Crashlytics CI friendly

naXa777 committed:
remove ignored files from VCS

naXa777 committed:
Modify build script to proceed with Lint errors (error is caused by com.googlecode.mp4parser dependency)

naXa777 committed:
Configure instrumented tests

naXa777 committed:
Embed build status image (badge)

naXa777 committed:
Update README.md - rearrange badges

naXa777 committed:
Start emulator before tests in order to fix Travis CI error: no connected devices

naXa777 committed:
Fix Travis CI error: could not parse .travis.yml (syntax error)

naXa777 committed:
Fix Travis CI error: Invalid --abi armeabi-v7a for the selected target

naXa777 committed:
Fix Travis CI error: Target id is not valid

naXa777 committed:
Fix Travis CI error: Invalid --abi armeabi-v7a for the selected target (try API level 21)

naXa777 committed:
Fix Travis CI error: remove -no-skin -no-audio + try API level 22 + cleanup travis.yml

naXa777 committed:
Increase timeout in order to fix Travis CI error: ShellCommandUnresponsiveException > InstallException > DeviceException

naXa777 committed:
Fix Travis CI error (attempt №10) - copy travis.yml from andstatus/andstatus

naXa777 committed:
Fix Travis CI error (attempt №11): INSTRUMENTATION_FAILED: Unable to find instrumentation info

naXa777 committed:
Fix Travis CI error (attempt №12): INSTRUMENTATION_FAILED: Unable to find instrumentation info

naXa777 committed:
Enable test coverage for debug build
Enable R8 shrinker for release build

This should fix Travis CI error related to coverage.ec: No such file or directory.

naXa777 committed:
Update README.md

include Codacy badge in the project description

naXa777 committed:
Update Crowdin configuration file

naXa777 committed:
Update README.md

Contribution guide

naXa777 committed:
Update README.md

More status badges!

naXa777 committed:
Update README.md

More status badges!!

naXa777 committed:
Upload coverage report to Codecov after successful build in Travis CI

naXa777 committed:
:bug: bugfix: duration is displayed as 00:00

naXa777 committed:
Add LeakCanary - a memory leak detection library

naXa777 committed:
New Crowdin translations (#34)

* New languages: Belarusian, Ukrainian
* New translations strings.xml: Spanish, French, Lithuanian, Russian

naXa777 committed:
Update README.md

 * Contribute automated tests
 (cherry-picked from gh-pages branch)

naXa777 committed:
Update README.md - 2 new badges

naXa777 committed:
Fix #35 Prevent mis-clicks on buttons

naXa777 committed:
Fix #21 Chronometer starts even if recording start failed

naXa777 committed:
#37 fix possible memory leaks
#35 Prevent misclicks on cards

naXa777 committed:
Enhance crash reports

naXa777 committed:
Fix #5 Activity has leaked ServiceConnection that was originally bound here

naXa777 committed:
update Gradle

naXa777 committed:
:busts_in_silhouette: Contributing guidelines

naXa777 committed:
try to fix anchor link

Commit:
Create strings.xml for South Korea

naXa777 commented:
hey @gaul and @dkim0419,

could you review and merge this PR?

gaul commented on app/app.iml:
```diff
@@ -62,59 +62,46 @@
+    <orderEntry type="library" exported="" name="Gradle: com.melnykov:floatingactionbutton-1.1.0" level="project" />
+    <orderEntry type="library" exported="" name="Gradle: com.android.support:appcompat-v7-21.0.3" level="project" />
+    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-v4-21.0.3" level="project" />
+    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-annotations:21.0.3@jar" level="project" />
```
These seem like spurious changes?

gaul commented on app/src/main/res/values/strings.xml:
```diff
@@ -36,7 +38,7 @@
 
     <string name="no_saved_recordings_message">No saved recordings</string>
-    <string name="pause_recording_button">Pause</string>
+    <string name="pause_recording_button">Paused</string>
```
Should this be Pause or Paused?

Jantzilla committed:
Add new options icon to sound file list item

naXa777 committed:
update Gradle and other dependencies

Jantzilla committed:
Change "presentFileOptions()" return value from "false" to "true"

naXa777 committed:
fix Travis CI pipeline :construction_worker: (#57)

* fix Travis CI pipeline :construction_worker:
* Update build tools to 29.0.0

har2008preet committed:
*.tmp files are now stored in external cache (#59)

fixes issue #25 

* `*.tmp` file is now stored in /storage/emulated/0/Android/data/by.naxa.soundrecorder/cache/SoundRecorder.by/*
* new method `createDirectory` is created in `Paths` to create directory
* optimized imports

Tgo1014 committed:
Added Adaptive Icons

naXa777 committed:
Merge pull request #62 from Tgo1014/feature/adaptative-icons

Added Adaptive Icons

Commit:
Successfully added night mode

msolanik committed:
Added displaying "No recordings" when record list is empty

naXa777 committed:
update Gradle wrapper to 5.6.3

naXa777 committed:
migrate Fabric Crashlytics to Firebase Crashlytics

naXa777 committed:
fix Travis CI pipeline :construction_worker:

naXa777 committed:
fix Travis CI pipeline :construction_worker:
add template google-services.json

naXa777 committed:
fix Travis CI pipeline :construction_worker:
add leakcanary library to `dev` build type

naXa777 committed:
fix Travis CI pipeline :construction_worker: (again)

naXa777 committed:
fix Travis CI pipeline :construction_worker: (again and again)
use template google-services.json

naXa777 committed:
Merge pull request #64 from MilindAmrutkar/issue-2

Successfully added night mode

naXa777 committed:
Merge remote-tracking branch 'origin/master' into maintenance/migrate-to-firebase

# Conflicts:
#	app/src/main/java/by/naxa/soundrecorder/SoundRecorderApplication.java

naXa777 committed:
add missing dependency

naXa777 committed:
Merge pull request #66 from naXa777/maintenance/migrate-to-firebase

Migrate to Firebase Crashlytics

naXa777 committed:
downgrade target SDK to P (not ready for Q yet)

aman0408 committed:
Added intent for other apps to use recorder (#63)

fixes #38
* add intent for other apps to use the recorder

naXa777 committed:
add translations from POEditor:

* Arabic (EG) by AhmadAyman99
* Persian by mahdi takrim

hyperov committed:
update gradle

hyperov committed:
add gradle properties

hyperov committed:
manual pull request from youcef934

hyperov committed:
code clean up

hyperov committed:
lint fixes
